strings/longest_substr_no_repeating.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file
+ * @brief Sliding window with an array to find the longest substring with no repeating characters.
+ * @details
+    * Given a string, find the longest substring with no repeating any characters.
+    * The time complexity is O(n) due to the sliding window method 
+    * and the space complexity is O(1) because the array is fixed size of 256.
+ * @author [Tan Nguyen](https://github.com/tazyqui)
+ */
+
+#include <cassert>   /// for assert
+#include <string>    // for string
+#include <algorithm> // for max
+
+/**
+ * @brief This function implement the sliding window technique along side an array in order to find
+ * the longest substring with no repeating character.
+ * @param String s, the given string to find the substring.
+ * @returns Return an integer that indicates the size of the longest substring with no repeating characters.
+ */
+int lengthOfLongestSubstring(std::string s) {   
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂